integration-tests/pkg/collector/collector_k8s.go

```diff
@@ -179,12 +200,28 @@ func (k *K8sCollectorManager) IsRunning() (bool, error) {
 }
 
 func (k *K8sCollectorManager) ContainerID() string {
-	cf := executor.ContainerFilter{
-		Name:      "collector",
-		Namespace: TEST_NAMESPACE,
+	if k.containerID == "" {
+		cf := executor.ContainerFilter{
+			Name:      "collector",
+			Namespace: TEST_NAMESPACE,
+		}
+
+		k.containerID = k.executor.PodContainerID(cf)
 	}
+	return k.containerID
+}
 
-	return k.executor.PodContainerID(cf)
+func (k *K8sCollectorManager) IP() string {
```

Comment:
This could be moved to k8s/base.go and it could take the pod name as a parameter rather than hard coding "collector". Then here we would just have a one line function that calls IP("collector").

Reply:
We need an executor to do this, so rather than `pkg/k8s` I would consider moving it to the k8s specific executor.

Reply:
Sounds good.